hugegraph-api/src/main/java/com/baidu/hugegraph/api/traversers/PredictionAPI.java

```diff
@@ -0,0 +1,122 @@
+ */
+
+package com.baidu.hugegraph.api.traversers;
+
+import static com.baidu.hugegraph.traversal.algorithm.HugeTraverser.DEFAULT_ELEMENTS_LIMIT;
+import static com.baidu.hugegraph.traversal.algorithm.HugeTraverser.DEFAULT_MAX_DEGREE;
+
+import javax.inject.Singleton;
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Context;
+
+import com.baidu.hugegraph.HugeGraph;
+import com.baidu.hugegraph.api.API;
+import com.baidu.hugegraph.api.graph.EdgeAPI;
+import com.baidu.hugegraph.api.graph.VertexAPI;
+import com.baidu.hugegraph.backend.id.Id;
+import com.baidu.hugegraph.core.GraphManager;
+import com.baidu.hugegraph.traversal.algorithm.PredictionTraverser;
+import com.baidu.hugegraph.type.define.Directions;
+import com.baidu.hugegraph.util.E;
+import com.baidu.hugegraph.util.JsonUtil;
+import com.codahale.metrics.annotation.Timed;
+import com.google.common.collect.ImmutableMap;
+
+/**
+ * This API include similar prediction algorithms, now include:
+ *  - Adamic Adar
+ *  - Resource Allocation
+ *
+ * Could add more prediction algorithms in future
+ */
+@Path("graphs/{graph}/traversers/")
+@Singleton
+public class PredictionAPI extends API {
+
+    @GET
+    @Timed
+    @Path("adamicadar")
+    @Produces(APPLICATION_JSON_WITH_CHARSET)
+    public String get(@Context GraphManager manager,
+                      @PathParam("graph") String graph,
+                      @QueryParam("vertex") String current,
+                      @QueryParam("other") String other,
+                      @QueryParam("direction") String direction,
+                      @QueryParam("label") String edgeLabel,
+                      @QueryParam("max_degree")
+                      @DefaultValue(DEFAULT_MAX_DEGREE) long maxDegree,
+                      @QueryParam("limit")
+                      @DefaultValue(DEFAULT_ELEMENTS_LIMIT) long limit) {
+        LOG.debug("Graph [{}] get adamic adar between '{}' and '{}' with " +
+                  "direction {}, edge label {}, max degree '{}' and limit '{}'",
+                  graph, current, other, direction, edgeLabel, maxDegree,
+                  limit);
+
+        Id sourceId = VertexAPI.checkAndParseVertexId(current);
+        Id targetId = VertexAPI.checkAndParseVertexId(other);
+        E.checkArgument(!current.equals(other),
+                        "The source and target vertex id can't be same");
+        Directions dir = Directions.convert(EdgeAPI.parseDirection(direction));
+
+        HugeGraph g = graph(manager, graph);
+        PredictionTraverser traverser = new PredictionTraverser(g);
+        double score = traverser.adamicAdar(sourceId, targetId, dir,
+                                            edgeLabel, maxDegree, limit);
+        return JsonUtil.toJson(ImmutableMap.of("adamic_adar", score));
+    }
+
+    @GET
+    @Timed
+    @Path("resourceallocation")
```

Comment:
prefer to split into 2 api files

Reply:
the prediction api have many similar implements, refer [here](https://neo4j.com/docs/graph-data-science/current/algorithms/linkprediction/)

seems redundant to divide them?

Reply:
The current style each algorithm interface is a separate api, it is only placed in two files, and there will be no extra code.